pipeline/transformer/transformer_tei/src/main/scala/weco/pipeline/transformer/tei/TeiXml.scala

```diff
@@ -70,7 +71,50 @@ class TeiXml(val xml: Elem) extends Logging {
     }
   }
 
-  def nestedTeiData: Either[Throwable, List[TeiData]] =
+  /**
+    * TEI works can be composed of other works.
+    * This function extracts the information about these nested works.
+    *
+    * Nested works can be specified in TEI as msItem or msPart depending
+    * if the manuscript is a single part manuscript or a multipart manuscript.
+    * check https://github.com/wellcomecollection/wellcome-collection-tei/blob/main/docs/TEI_Manual_2020_V1.pdf
+    * for more info.
+    */
+  def nestedTeiData = nestedTeiDataFromItems.flatMap {
+    case Nil      => title.flatMap(nestedTeiDataFromParts)
+    case teiDatas => Right(teiDatas)
+  }
+
+  /**
+    * Extract information about inner works for multi part manuscripts.
+    * Multi part manuscripts have msPart elements containing information about inner works.
+    * msParts don't have a title so we construct the title concatenating the
+    * title of the wrapper work and the part number.
+    */
+  private def nestedTeiDataFromParts(
+    wrapperTitle: String): Either[Throwable, List[TeiData]] =
+    (xml \\ "msDesc" \ "msPart")
+      .map { node =>
+        for {
+          id <- getIdFrom(node)
+          partNumber <- Try((node \@ "n").toInt).toEither
+          description <- summary(node \ "summary")
+          languages <- TeiLanguages.parseLanguages(node)
+        } yield
+          TeiData(
+            id = id,
+            title = s"$wrapperTitle part $partNumber",
+            languages = languages,
+            description = description)
+      }
+      .toList
+      .sequence
+
+  /**
+    * Extract information about inner works for single part manuscripts.
```

Comment:
Can you explain what a single part manuscript is? Or perhaps point to an example of one?

Reply:
In the comment on the public function for nested data there's a link to the tei manual. I think that's the best source of information rather than me trying to rephrase it here